src/amqpvalue.c

```diff
@@ -4959,6 +4969,7 @@ static int internal_decoder_decode_bytes(INTERNAL_DECODER_DATA* internal_decoder
                 {
                     AMQP_VALUE_DATA* descriptor;
                     internal_decoder_data->decode_to_value->type = AMQP_TYPE_DESCRIBED;
+                    internal_decoder_data->decode_to_value->value.described_value.value = NULL;
```

Comment:
If this is not NULL already, its value should be REF COUNT DEC before nulling it out, otherwise we might have a memory leak. Please check where it gets set, on 
https://github.com/Azure/azure-uamqp-c/pull/376/files#diff-7ec5b31fa5071f1609135ee9c21730636a478036d89115c51781e7353fdcbaf0R5435

Reply:
moved zeroing out the struct where it's allocated.